gdal/frmts/esric/esric_dataset.cpp

```diff
@@ -0,0 +1,446 @@
+        if (0 == size)
+            return MissingTile(nBlockXOff, nBlockYOff, pData);
+        auto& fbuffer = parent->filebuffer;
+        fbuffer.resize(size_t(size));
+        VSIFSeekL(bundle.fh, offset, SEEK_SET);
+        if (size != VSIFReadL(fbuffer.data(), size_t(1), size_t(size), bundle.fh)) {
+            CPLError(CE_Failure, CPLE_FileIO, "Error reading tile");
+            return CE_Failure;
+        }
+        CPLString magic;
+        // Should use some sort of unique
+        magic.Printf("/vsimem/esric_%u.tmp", rand());
+        auto mfh = VSIFileFromMemBuffer(magic.c_str(), fbuffer.data(), size, false);
+        VSIFCloseL(mfh);
+        // Can't open a raster by handle?
+        auto inds = GDALOpen(magic.c_str(), GA_ReadOnly);
+        if (!inds) {
+            VSIUnlink(magic.c_str());
+            CPLError(CE_Failure, CPLE_FileIO, "Error opening tile");
+            return CE_Failure;
+        }
+        // Duplicate first band if not sufficient bands are provided
+        auto inbands = GDALGetRasterCount(inds);
+        int ubands[4] = { 1, 1, 1, 1 };
+        int* usebands = nullptr;
+        int bandcount = parent->nBands;
+        if (inbands != bandcount) {
+            // Opaque if output expects alpha channel
+            if (0 == bandcount % 2) {
+                fill(buffer.begin(), buffer.end(), GByte(255));
+                bandcount--;
+            }
+            if (3 == inbands) {
+                // Lacking opacity, copy the first three bands
+                usebands = ubands;
+                ubands[1] = 2;
+                ubands[2] = 3;
+            }
+            else if (1 == inbands) {
+                // Grayscale, expecting color
+                usebands = ubands;
+            }
+        }
+
+        auto errcode =
+            GDALDatasetRasterIO(inds, GF_Read, 0, 0, TSZ, TSZ, buffer.data(), TSZ, TSZ, GDT_Byte,
+                bandcount, usebands, parent->nBands, parent->nBands * TSZ, 1);
+        GDALClose(inds);
+        VSIUnlink(magic.c_str());
+        // Error while unpacking tile
+        if (CE_None != errcode)
+            return errcode;
+
+        for (int iBand = 2; iBand <= parent->nBands; iBand++)
+        {
+            auto band = parent->GetRasterBand(iBand);
+            if (lvl)
+                band = band->GetOverview(lvl - 1);
+            auto poBlock = band->GetLockedBlockRef(nBlockXOff, nBlockYOff);
+            if (poBlock != nullptr)
+                poBlock->DropLock();
+        }
+    }
+
+    GDALCopyWords(buffer.data() + nBand - 1, GDT_Byte, parent->nBands, pData, GDT_Byte, 1, TSZ * TSZ);
+    return CE_None;
+} // IReadBlock
+
+} // namespace
+
+#define ENDS_WITH_CI(a,b) (strlen(a) >= strlen(b) && EQUAL(a + strlen(a) - strlen(b), b))
+
+// Without full XML parsing, weak, might still fail
+static int Identify(GDALOpenInfo* poOpenInfo) {
+    if (poOpenInfo->eAccess != GA_ReadOnly
+        || !ENDS_WITH_CI(poOpenInfo->pszFilename, "conf.xml")
```

Comment:
You can surround this tests that checks filename ends with conf.xml by the following
```
#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
 || !ENDS_WITH_CI(poOpenInfo->pszFilename, "conf.xml")
#endif
```
so that fuzzers don't care about filename